### ISSUE TITLE
gray out "Update Layer with External Edits" when no file is associated

### ISSUE THREAD

**Issue by jeisner** [mypaint/mypaint#580]
If the external edit command has never been invoked on a layer (during the current session), then there is no file associated with that layer.  

The "update layer" command is inapplicable in this case (it does nothing) and should be grayed out.

**Comment by achadwick** [comment]
Sounds reasonable. It's just another action to update when the user switches layer.

The only UI question one has to ask is whether the reason for the greying-out would ever be a mystery to the user. Style guides tend to say that one should leave the menu entry ungreyed when it isn't obvious how to ungrey it, and just have the command do nothing.

I think it would always be presented under commands for starting external edits, so probably it would be pretty obvious.

**Comment by jeisner** [comment]
Agree that the concern was worth raising and that it can be dismissed in this case for the reason you give.

**Comment by odysseywestra** [comment]
An alternate thing we could do is add an icon to the layers which have an external app open on them. Here's a Mockup:

![external_example](https://cloud.githubusercontent.com/assets/1840562/12808554/862ac718-cacc-11e5-9dc6-c2553d6b90e4.png)

Also it could be programmed that if you click on that icon, the external application will be focused.

BTW I did grab the icon from [Font Awesome](http://fortawesome.github.io/Font-Awesome/icon/external-link/) Website if you want to add that icon to the symbolic icons we have.

**Comment by achadwick** [comment]
> it could be programmed that if you click on that icon, the external application will be focused. 

Not so sure that this will be possible. Can Gio/GApplication do that for us?

The icon looks easy enough to recreate with a non-font license in our own symbolics.

**Comment by odysseywestra** [comment]
@achadwick yeah I can easily recreate it. that shouldn't be a problem. With making it like a button, I really have no idea. I'll have to look into it.
